### PR TITLE
Responses error handling - dialogs mainly

### PR DESCRIPTION
**Pull request by thiessenp-cds**
# Summary | Résumé

Updates the content for the responses dialog error messages.

# Test instructions | Instructions pour tester la modification

Go to the responses page of a published form with at least a few submissions. Try to generate the errors. The server error will probably be only possible by locally setting this variable in `DownloadTableDialog.tsx`:
from
```
  const [errors, setErrors] = useState({
    minEntries: false,
    maxEntries: false,
    errorEntries: false,
    invalidEntry: false,
    unknown: false,
  });
```
to
```
  const [errors, setErrors] = useState({
    minEntries: false,
    maxEntries: false,
    errorEntries: false,
    invalidEntry: false,
    unknown: true,
  });
```
You can also test all the errors by setting the above values to true/false but that may not be a true test :)

Also the errors are in a Figma design draft from Anik: https://www.figma.com/file/6QAmUEq6d8tyXzWDGKO78s/v2.-Form-builder?node-id=5346-72222&t=kGjpcC6a3fKfTnXU-4

Here are the errors for convenience in French (because it's Wednesday :)
*Confirm Dialog*
Screenshot 2023-04-12 at 9 56 44 AM
Screenshot 2023-04-12 at 9 58 40 AM
Screenshot 2023-04-12 at 9 59 02 AM
Screenshot 2023-04-12 at 10 04 22 AM
Screenshot 2023-04-12 at 10 05 07 AM

*Report Dialog*
Screenshot 2023-04-12 at 10 08 27 AM
Screenshot 2023-04-12 at 10 08 47 AM
Screenshot 2023-04-12 at 10 35 45 AM
Screenshot 2023-04-12 at 10 36 22 AM
Screenshot 2023-04-12 at 10 05 46 AM

# Unresolved questions / Out of scope | Questions non résolues ou hors sujet

*Known Bugs*
- Currently the Report dialog will show success when at least one report form number is valid, even if there are N-more invalid reports. This should be resolved when/if the report API is updated.
- A file download or dialog update for confirm/problem may or may not update the table data. I tried adding a delay and this worked well for a time but it seems to be working less consistently now. There is an existing issue for this https://github.com/cds-snc/platform-forms-client/issues/1744

*Refactoring*
- Would be nice if the report and confirm API's both returned a list of invalid entries. Currently only the confirm API does
- Move files to more logical locations e.g. line-item-entries/ is currently in a shared components directory but is very coupled to the responses dialog. It should either be moved to the responses/ dir or refactored to be generic/reusable 
- A single dialog was reused for the two dialogs. This was to share what started as the same behaviour. As time went on the dialogs became more different and the number of params passed in is a bit unmaintainable. These should probably be separated into two in a later PR

*Usability Related*
- Currently it is possible for a dialog to show more than one error (just set the errors object to all true :). This is an unlikely case but if it becomes a usability problem or anyone feels only one error should be shown at a time, it should be an easy fix
- Showing some kind of success message for a successful dialog confirmation/problem may make it more clear the action was successful
- Probably more error messages e.g. a user trying to update form s/he doesn't have access to
- The usability for the dialogs could probably be improved. Such as how entries are edited, submitted.. 
- More accessibility testing, especially Jaws+Edge

# Pull Request Checklist

Please complete the following items in the checklist before you request a review:

- [x] Have you completely tested the functionality of change introduced in this PR? Is the PR solving the problem it's meant to solve within the scope of the related issue?
- [x] The PR does not introduce any new issues such as failed tests, console warnings or new bugs.
- [ ] If this PR adds a package have you ensured its licensed correctly and does not add additional security issues?
- [x] Is the code clean, readable and maintainable? Is it easy to understand and comprehend.
- [ ] Does your code have adequate comprehensible comments? Do new functions have [docstrings](https://tsdoc.org/)?
- [ ] Have you modified the change log and updated any relevant documentation?
- [ ] Is there adequate test coverage? Both unit tests and end-to-end tests where applicable?
- [x] If your PR is touching any UI is it accessible? Have you tested it with a screen reader? Have you tested it with automated testing tools such as axe?

### CODE REVIEW

File: public/static/locales/en/form-builder-responses.json [hunk 1]
```diff
@@ -43,54 +43,86 @@
...
+      "downloadingXFiles": "Downloading {{fileCount}} files"
+    },
+    "errors": {
       "trySelectingLessFilesHeader": "Only {{max}} responses can be downloaded at a time.",
       "trySelectingLessFiles":"You have {{max}} responses selected. Uncheck at least 1 response before downloading.",
       "atLeastOneFileHeader": "At least one file must be selected to download.",
       "atLeastOneFile": "Try selecting at least one file to download.",
       "errorDownloadingFilesHeader":"Error downloading the selected files.",
       "errorDownloadingFiles":"Try again in a few minutes. If the problem persists contact support."
     },
```
Comment (timarney) [inline public/static/locales/en/form-builder-responses.json:56]: "errorDownloadingFilesLink" missing?

Comment (thiessenp-cds) [reply public/static/locales/en/form-builder-responses.json:56]: Good find. I should also double check that error since I wrote it I think
